swift-sdk/Internal/IterableEmbeddedManager.swift

```diff
@@ -43,8 +50,58 @@ class IterableEmbeddedManager: NSObject, IterableEmbeddedManagerProtocol {
     public func syncMessages(completion: @escaping () -> Void) {
         retrieveEmbeddedMessages(completion: completion)
     }
+    
+    public func handleEmbeddedClick(message: IterableEmbeddedMessage?, buttonIdentifier: String?, clickedUrl: String) {
+        print("called embeddedMessageClicked IterableEmbeddedManager method.")
+        guard let message = message else {
+            print("Error: message is nil.")
+            return
+        }
+
+        if let url = URL(string: clickedUrl) {
+            handleClick(clickedUrl: url, forMessage: message)
+        } else {
+            print("Invalid URL: \(clickedUrl)")
+        }
+    }
+    
+    private func handleClick(clickedUrl url: URL, forMessage message: IterableEmbeddedMessage) {
+        
+        handleUrl(url: url.absoluteString)
+    }
+    
+    
+    private func createAction(fromUrlOrAction url: String) -> IterableAction? {
```

Comment:
Since this is copied from the in-app manager code, maybe it'd be better to move this stuff to some kind of a helper class to reduce duplication.